lib/fa2/asset/multi_asset.impl.jsligo

```diff
@@ -0,0 +1,263 @@
+#import "../common/errors.mligo" "Errors"
+
+#import "../common/tzip12.datatypes.jsligo" "TZIP12"
+
+#import "../common/tzip12.interfaces.jsligo" "TZIP12Interface"
+
+#import "../common/tzip16.datatypes.jsligo" "TZIP16"
+
+export namespace MultiAsset implements TZIP12Interface.FA2{
+   export type Ledger = big_map<[address, nat], nat>;
+   type operator = address;
+   export type Operators = big_map<[address, operator], set<nat>>;
+   export type storage = {
+      ledger: Ledger,
+      operators: Operators,
+      token_metadata: TZIP12.tokenMetadata,
+      metadata: TZIP16.metadata
+   };
+   type ret = [list<operation>, storage];
+   //export namespace Operators {
+   // if transfer policy is Owner_or_operator_transfer
+
+   export const assert_authorisation = (
+      [operators, from_, token_id]: [Operators, address, nat]
+   ): unit => {
+      const sender_ = (Tezos.get_sender());
+      if (sender_ != from_) {
+         const authorized =
+            match((Big_map.find_opt([from_, sender_], operators))) {
+               when (Some(a)):
+                  a
+               when (None()):
+                  Set.empty
+            };
+         if (! (Set.mem(token_id, authorized))) {
+            return failwith(Errors.not_operator)
+         }
+      }
+   };
+   const assert_update_permission = (owner: address): unit =>
```

Comment:
Asserts should be exported and in another file namespace Assert ?

Reply:
I removed already inner namespaces as it became unreadable

Reply:
I mean, it should be in another file